dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -393,4 +394,51 @@ public CodeBlock multimapParam() {
             }
         });
     }
+
+    private CodeBlock generateListElementSerializerCall(CodeBlock argName, ArgumentType type) {
+        CodeBlock elementName = CodeBlock.of("$L$L", argName, "Element");
+        return type.match(new ArgumentType.Cases<>() {
+            @Override
+            public CodeBlock primitive(TypeName _typeName, String parameterSerializerMethodName) {
+                return CodeBlock.of(
+                        ".$L($L -> $L.$L($L))",
+                        MAP,
```

Comment:
A couple minor nits: I think we're better off inlining the `.map(` literal for readability, and removing the constant. Javapoet prefers to use `$N` for names ([docs](https://github.com/square/javapoet?tab=readme-ov-file#n-for-names)). Positional arguments may be helpful to reference the same component twice (certainly not always something we want to do, as it can harm readability) [docs](https://github.com/square/javapoet?tab=readme-ov-file#positional-arguments).

```suggestion
                        ".map($1N -> $2N.$3N($1N))",
```

Reply:
Isn't this equivalent to `".map($2N::$3N)"`? Using a method reference is a bit easier to read when possible

Reply:
Fixed